nilearn/regions/connectivity.py

```diff
@@ -0,0 +1,392 @@
+"""Methods For defining voxelwise connectivity"""
+# Author: Derek Pisner
+
+import numpy as np
+from scipy import divide, prod
+from .._utils.niimg import _safe_get_data
+
+def indx_1dto3d(idx, sz):
+    """
+    Translate 1D vector coordinates to 3D matrix coordinates for a 3D matrix
+    of size sz.
+
+    Parameters
+    ----------
+    idx : array
+        A 1D numpy coordinate vector.
+    sz : array
+        Shape of 3D matrix idx.
+
+    Returns
+    -------
+    x : int
+        x-coordinate of 3D matrix coordinates.
+    y : int
+        y-coordinate of 3D matrix coordinates.
+    z : int
+        z-coordinate of 3D matrix coordinates.
+    """
+
+    x = divide(idx, prod(sz[1:3]))
+    y = divide(idx - x * prod(sz[1:3]), sz[2])
+    z = idx - x * prod(sz[1:3]) - y * sz[2]
+    return x, y, z
+
+
+def indx_3dto1d(idx, sz):
+    """
+    Translate 3D matrix coordinates to 1D vector coordinates for a 3D matrix
+    of size sz.
+
+    Parameters
+    ----------
+    idx : array
+        A 3D numpy array of matrix coordinates.
+    sz : array
+        Shape of 3D matrix idx.
+
+    Returns
+    -------
+    idx1 : array
+        A 1D numpy coordinate vector.
+    """
+
+    if np.linalg.matrix_rank(idx) == 1:
+        idx1 = idx[0] * prod(sz[1:3]) + idx[1] * sz[2] + idx[2]
+    else:
+        idx1 = idx[:, 0] * prod(sz[1:3]) + idx[:, 1] * sz[2] + idx[:, 2]
+    return idx1
+
+
+def make_local_connectivity_scorr(func_img, clust_mask_img, thresh):
+    """
+    Constructs a spatially constrained connectivity matrix from a fMRI dataset.
+    The weights w_ij of the connectivity matrix W correspond to the
+    spatial correlation between the whole brain FC maps generated from the
+    time series from voxel i and voxel j. Connectivity is only calculated
+    between a voxel and the 27 voxels in its 3D neighborhood
+    (face touching and edge touching).
+
```

Comment:
We could call it that sure. I called it connectivity since its use-case is to fill the connectivity argument of AgglomerativeClustering (which is called connectivity). Conceivably these functions could also be condensed into a single class, called make_local_correlation, with scorr, tcorr, and ones (i.e. grid_to_graph) as methods if you think that'd be cleaner?